### ISSUE TITLE
Dialog closed not executing

### ISSUE THREAD

**Issue by swesley** [foex-open-source/fos-redirect#1]
I have a test case that modifies the Sample Dialog application to include a column that opens the dialog using the FOS - Redirect plugin.
[https://apex.oracle.com/pls/apex/swesley/r/18627/page-modal-dialog](https://apex.oracle.com/pls/apex/swesley/r/18627/page-modal-dialog)
I would have expected the original Dialog Closed event on the region, but it does not.

I've used onClick on the new column, set the value of a hidden item using 
`$(this.triggeringElement).data('deptno')`
Then open the same p7 using the FOS DA, using the option to submit the P4_DEPTNO before redirect to the static URL.

Have I missed a step, or is this expected behaviour? 
Cheers

**Comment by swesley** [comment]
I believe this can be corrected by adding attribute 11 to represent the triggering element within the get_url function.

```
    -- triggering element that is invoking redirect.
    -- use this so relevant Dialog Closed actions will execute
    l_triggering_element    varchar2(4000)                           := p_dynamic_action.attribute_11;

...

        l_url := apex_util.prepare_url
           (l_url
           ,p_checksum_type => 'SESSION'
           ,p_triggering_element => l_triggering_element);
```

I've forked the repository and applied my changes, for reference. The positional reference for the checksum type parameter was also incorrect.

**Comment by mnolan866** [comment]
Thanks Scott for raising this, and apologies for the late reply. It seems we missed the automatic notification for this issue. 

We will look at trying to avoid the extra attribute by doing a dictionary lookup on the triggering element. 

Looks like we missed the prepare_url call issue during testing, good catch. 

We will address both things in the next release.

**Comment by alanarentsen** [comment]
Matt,
We needed this. I have implemented it without using an extra parameter. You can check it out in my fork.
Hope you can merge it.
Thanks

**Comment by alanarentsen** [comment]
Oh I didn't export the Plugin so the change has to be done over there too. I didn't know in which version you want to do the export.

**Comment by mnolan866** [comment]
I can confirm that we will use Alan's fix in the next FOS 21.1 release due out soon.

**Comment by foex-open-source** [comment]
Fixed in just published v21.1.0
